### PR TITLE
update travis to run markdown linter remark

### PR DESCRIPTION
**Pull request by dirk-thomas**
I will comment on each configuration option why I chose it.

### CODE REVIEW

**Comment by wjwwood** [comment]
Lgtm

**Comment by tfoote** [comment]
Does this replace #67?

File: .remarkrc [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "plugins": {
+    "remark-lint": {
+      "fenced-code-flag": {
+        "allowEmpty": true
```
Comment (dirk-thomas) [inline .remarkrc:5]: The current files contain code snippets without a language.

File: .remarkrc [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+      },
+      "list-item-indent": "space",
```
Comment (dirk-thomas) [inline .remarkrc:7]: We want list items to have a single space between the marker and the text.

File: .remarkrc [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+      "maximum-heading-length": 999,
+      "maximum-line-length": 9999,
```
Comment (dirk-thomas) [inline .remarkrc]: We allow "unlimited" line length.

File: .remarkrc [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+      "no-file-name-irregular-characters": false,
```
Comment (dirk-thomas) [inline .remarkrc:10]: The current files use underscores for separation.

File: .remarkrc [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+      "no-html": false,
```
Comment (dirk-thomas) [inline .remarkrc:11]: The current files use `<div class="abstract" markdown="1">`.

File: .remarkrc [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+      "unordered-list-marker-style": "-"
```
Comment (dirk-thomas) [inline .remarkrc:12]: Following http://www.cirosantilli.com/markdown-style-guide/#lists

**Comment by dirk-thomas** [comment]
I have not yet enabled `mdast-lint-sentence-newline` since it currently results in many false positives (see https://github.com/chcokr/mdast-lint-sentence-newline/issues).

**Comment by dirk-thomas** [comment]
Replaced label "in progress" with "in review".

File: .remarkrc (hunk 3 above)
Comment (wooorm) [inline .remarkrc]: You should be able to pass `false` to ignore this rule, which looks neater than specifying a large number :)

Comment (dirk-thomas) [inline .remarkrc]: Good point, thanks: b5985fe11fbd5af76d00997d983eb956818b2526

**Comment by jackieokay** [comment]
+1. Optional, but consider adding a note in `contributing.md` about the linter and Markdown style guidelines.

**Comment by dirk-thomas** [comment]
I added a note in 45500528be2e58438536d0d406da3ecd42942415

**Comment by dirk-thomas** [comment]
Since @jacquelinekay gave a +1 I will now rewrite some commits and then go ahead with the merge after Travis is still green.

**Pull request merged**